code/game/data_huds.dm

```diff
@@ -193,11 +194,20 @@
 		if(HAS_TRAIT(src, TRAIT_XENO_HOST))
 			holder.icon_state = "hudxeno"
 		else if(stat == DEAD)
+			if(!getorgan(/obj/item/organ/brain) || soul_departed() || ishellbound())
+				holder.icon_state = "huddead-permanent"
+				return
 			if(tod)
 				var/tdelta = round(world.time - timeofdeath)
 				if(tdelta < (DEFIB_TIME_LIMIT * 10))
+					if(!client && key)
+						holder.icon_state = "huddefib-ssd"
+						return
 					holder.icon_state = "huddefib"
 					return
+			if(!client && key)
+				holder.icon_state = "huddead-ssd"
+				return
```

Comment:
Since we have new conditions, we need to call med_hud_set_status when hellbound is set, when soul departed changes and when the brain condition changes.
SSD indicator seems well handled.

Reply:
Problem with soul departed is that the code for transferring a mind is... not hookable, at all.
It's all just `key` assignments.

Brain is already handled because removing/adding a brain to a body transfers the mind and thus calls `transfer_to` which I added update calls to.